### PR TITLE
feat: Create a Exporter class to allow export data from LandingLens

### PR DESCRIPTION
**Pull request by CamiloInx**
Motivation
---
* Create a `Exporter` class to provide the users with a set of methods to export data from LandingLens
* The first method introduced is the `export_event_logs()` that allows organization owners the ability to export the event logs of the organization from the given time range.

Test
---
Added unit test

Screenshot 2024-02-07 at 8 31 42 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206481707799922

### CODE REVIEW

File: landingai/data_management/export.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        self._client = LandingLens(project_id=project_id, api_key=api_key)
+
+    def export_event_logs(self, from_timestamp: str, to_timestamp: str) -> Dict[str, str]:
+        """Exports the event logs of the organization from the given time range.
+        Parameters
+        ----------
+        from_timestamp: str
+            In UTC following the format "YYYY-MM-DD HH:MM:SS.SSS"
+        to_timestamp: str
+            In UTC following the format "YYYY-MM-DD HH:MM:SS.SSS"
```
Comment (MingruiZhang) [inline landingai/data_management/export.py:36]: Asking user for timestamp seems like a bad experience, can we simplify this to date e.g. `2024-02-04` . And `to_timestamp` could be optional, default to today

File: landingai/data_management/export.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        """
+        # Check if from_timestamp and to_timestamp are in UTC format
+        try:
+            datetime.strptime(from_timestamp, "%Y-%m-%d %H:%M:%S.%f")
+            datetime.strptime(to_timestamp, "%Y-%m-%d %H:%M:%S.%f")
+        except ValueError:
+            raise ValueError("from_timestamp and to_timestamp must be in UTC format")
+
+        resp = self._client._api(EVENT_LOGS, params={"fromTimestamp": from_timestamp, "toTimestamp": to_timestamp})
+        resp_data = resp["data"]
```
Comment (MingruiZhang) [inline landingai/data_management/export.py:58]: If we have provided the signed url, we should directly help user download the csv.

Comment (CamiloInx) [reply landingai/data_management/export.py]: Done

File: landingai/data_management/export.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    def export_event_logs(self, from_date: str, save_path: str) -> None:
+        """Exports the event logs of the organization from the given time range.
+
+        Parameters
+        ----------
+        from_date: str
+            In date following the format "YYYY-MM-DD"
+        save_path:
+            Desired path to save the csv file to: '/path/to/save/file.csv'
```
Comment (MingruiZhang) [inline landingai/data_management/export.py:41]: optionally have to_date, if not provided, default to current time.

File: landingai/data_management/export.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        )
+        signed_url = resp["data"].get("signedUrl")
+        _LOGGER.debug("Signed URL: ", signed_url)
+        self._download_file_from_signed_url(signed_url, save_path)
+        print(f"Event logs exported successfully to path: {save_path}")
+
+    def _download_file_from_signed_url(self, signed_url: str, save_path: str) -> None:
+        response = requests.get(signed_url)
+        with open(save_path, "wb") as file:
+            file.write(response.content)
```
Comment (MingruiZhang) [inline landingai/data_management/export.py:70]: Let's just `webbrowser.open(signed_url)` to directly download the resource through the browser.

File: tests/unit/landingai/data_management/test_export.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+@freeze_time("2024-01-30 00:00:00.000000")  # Mock the current date and time
+def test_export_event_logs(mocked_method):
+    responses._add_from_file(
+        file_path="tests/data/responses/test_export_event_logs.yaml"
+    )
+    project_id = 12345
+    api_key = "land_sk_12345"
+    client = Exporter(project_id, api_key)
+    res = client.export_event_logs("2021-06-01", "./local/path/to/save/file.csv")
+    assert res is None
```
Comment (MingruiZhang) [inline tests/unit/landingai/data_management/test_export.py:19]: We can return the signed_url and check the return value

**Pull request closed without merging**